Dockerfile

```diff
@@ -1,25 +1,46 @@
-FROM python:3.9-alpine
+FROM python:3.9-alpine as buildimage
 
-VOLUME [ "/config"]
+COPY . /tmp/app_install
 
-# Install required dependencies
-RUN apk add --no-cache \
-# Support for Timezones
-    tzdata \
+RUN set -eux;\
+	apk add --no-cache \
 # ujson won't compile without these libs
-    g++
+		g++; \
+# wheel all packages for habapp
+	cd /tmp/app_install; \
+	pip wheel --wheel-dir=/root/wheels --use-feature=in-tree-build .
 
-# Always use latest versions
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
-COPY . .
+FROM python:3.9-alpine
 
-# Install from pip (we don't use that in the github actions because
-# the new version might not be available on pypi yet)
-# RUN pip3 install habapp
+COPY --from=buildimage /root/wheels /root/wheels
+COPY container/entrypoint.sh /entrypoint.sh
 
-# Install from checked out git branch
-RUN pip3 install .
+ENV HABAPP_HOME=/habapp \
+    USER_ID=9001 \
+    GROUP_ID=9001
+
+RUN set -eux; \
+# Install required dependencies
+ 	apk add --no-cache \
+    	bash \
+		su-exec \
+		tini \
+# Support for Timezones
+    	tzdata; \
+    mkdir -p ${HABAPP_HOME}; \
+    mkdir -p ${HABAPP_HOME}/config; \
+# install HABApp
+    pip3 install \
+        --no-index \
+        --find-links=/root/wheels \
+		habapp; \
```

Comment:
Can you please install from the sources and not from pip?

Reply:
I think you mean not to install from pypi?!

This is not the case here the complete local src is copied to the buildimage (line 3) and a wheel is build and put together with all dependencies wheels from pypi (line 11) Those are copied to /root/wheels of the final container (line 15) the  combination of --no-index  and --find-links tells pip to install from this local files not from pypi

Here is the log output of the command

```bash
+ pip3 install --no-index '--find-links=/root/wheels' habapp
Looking in links: /root/wheels
Processing /root/wheels/HABApp-0.31.3.dev0-py3-none-any.whl
Processing /root/wheels/aiohttp_sse_client-0.2.1-py2.py3-none-any.whl
Processing /root/wheels/paho_mqtt-1.6.1-py3-none-any.whl
Processing /root/wheels/ujson-5.0.0-cp39-cp39-musllinux_1_1_x86_64.whl
Processing /root/wheels/voluptuous-0.12.2-py3-none-any.whl
Processing /root/wheels/aiohttp-3.8.1-cp39-cp39-musllinux_1_1_x86_64.whl
Processing /root/wheels/stackprinter-0.2.5-py3-none-any.whl
Processing /root/wheels/EAScheduler-0.1.4-py3-none-any.whl
Processing /root/wheels/EasyCo-0.2.3-py3-none-any.whl
Processing /root/wheels/immutables-0.16-cp39-cp39-linux_x86_64.whl
Processing /root/wheels/pendulum-2.1.2-cp39-cp39-linux_x86_64.whl
Processing /root/wheels/pydantic-1.8.2-py3-none-any.whl
Processing /root/wheels/watchdog-2.1.6-py3-none-any.whl
Processing /root/wheels/bidict-0.21.4-py3-none-any.whl
Processing /root/wheels/frozenlist-1.2.0-cp39-cp39-musllinux_1_1_x86_64.whl
Processing /root/wheels/attrs-21.4.0-py2.py3-none-any.whl
Processing /root/wheels/multidict-5.2.0-cp39-cp39-musllinux_1_1_x86_64.whl
Processing /root/wheels/async_timeout-4.0.2-py3-none-any.whl
Processing /root/wheels/aiosignal-1.2.0-py3-none-any.whl
Processing /root/wheels/charset_normalizer-2.0.10-py3-none-any.whl
Processing /root/wheels/yarl-1.7.2-cp39-cp39-musllinux_1_1_x86_64.whl
Processing /root/wheels/astral-2.2-py2.py3-none-any.whl
Processing /root/wheels/pytz-2021.3-py2.py3-none-any.whl
Processing /root/wheels/ruamel.yaml-0.17.20-py3-none-any.whl
Processing /root/wheels/pytzdata-2020.1-py2.py3-none-any.whl
Processing /root/wheels/python_dateutil-2.8.2-py2.py3-none-any.whl
Processing /root/wheels/typing_extensions-4.0.1-py3-none-any.whl
Processing /root/wheels/six-1.16.0-py2.py3-none-any.whl
Processing /root/wheels/idna-3.3-py3-none-any.whl
Processing /root/wheels/ruamel.yaml.clib-0.2.6-cp39-cp39-linux_x86_64.whl
```